velox/dwio/common/ScanSpec.cpp

```diff
@@ -558,9 +558,12 @@ void filterRows(
 
 } // namespace
 
-void ScanSpec::applyFilter(const BaseVector& vector, uint64_t* result) const {
+void ScanSpec::applyFilter(
+    const BaseVector& vector,
+    vector_size_t size,
+    uint64_t* result) const {
   if (filter_) {
-    filterRows(vector, *filter_, vector.size(), result);
+    filterRows(vector, *filter_, size, result);
```

Comment:
do we handle the case where the `size` can be larger than child->size() ? I noticed  at L540-L542 we use the `vector.isNullAt(i)` which can be problematic.
can you also add a test case for this?